source/common/http/utility.cc

```diff
@@ -329,6 +329,29 @@ Utility::parseCookies(const RequestHeaderMap& headers,
   return cookies;
 }
 
+absl::InlinedVector<absl::string_view, 2> Utility::parseCookieValues(const HeaderMap& headers,
+                                                                     const absl::string_view key,
+                                                                     const size_t max_vals) {
+  absl::InlinedVector<absl::string_view, 2> ret;
+
+  forEachCookie(headers, Http::Headers::get().Cookie,
+                [&ret, &key, &max_vals](absl::string_view k, absl::string_view v) -> bool {
+                  if (k == key) {
+                    ret.push_back(v);
+                    // max_vals == 0 => infinity, so the condition above will never be true
+                    // in this case.
+                    if (ret.size() == max_vals) {
```

Comment:
If we default to std::numeric_limits::max instead of treating 0 as special I think this logic would seem a bit cleaner

Reply:
Done in https://github.com/envoyproxy/envoy/pull/21496/commits/b7f34796cf5bbe216a159745a0a86cc49a0d2268